### PR TITLE
Organization role management

### PR DESCRIPTION
**Pull request by kavinda1995**
* Organization Role List View
* Add Organization Role Modal
* Organization Role Edit View
* Fixes and improvements to overall organization management UIs

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/organizations/components/edit-organization-role/edit-organization-groups.tsx [hunk 1]
```diff
@@ -0,0 +1,714 @@
+/**
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (thivi) [inline apps/console/src/features/organizations/components/edit-organization-role/edit-organization-groups.tsx:2]:
```suggestion
 * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.com) All Rights Reserved.
```

File: apps/console/src/features/organizations/components/edit-organization-role/edit-organization-permission.tsx [hunk 2]
```diff
@@ -0,0 +1,160 @@
+/**
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (thivi) [inline apps/console/src/features/organizations/components/edit-organization-role/edit-organization-permission.tsx:2]:
```suggestion
 * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.com) All Rights Reserved.
```

File: apps/console/src/features/organizations/components/edit-organization-role/edit-organization-role-basic.tsx [hunk 3]
```diff
@@ -0,0 +1,434 @@
+/**
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (thivi) [inline apps/console/src/features/organizations/components/edit-organization-role/edit-organization-role-basic.tsx:2]:
```suggestion
 * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.com) All Rights Reserved.
```

File: apps/console/src/features/organizations/components/edit-organization-role/edit-organization-role-users.tsx [hunk 4]
```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (thivi) [inline apps/console/src/features/organizations/components/edit-organization-role/edit-organization-role-users.tsx:2]:
```suggestion
 * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.com) All Rights Reserved.
```

File: apps/console/src/features/organizations/components/edit-organization-role/edit-organization-role.tsx [hunk 5]
```diff
@@ -0,0 +1,159 @@
+/**
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (thivi) [inline apps/console/src/features/organizations/components/edit-organization-role/edit-organization-role.tsx:2]:
```suggestion
 * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.com) All Rights Reserved.
```

File: apps/console/src/features/organizations/components/edit-organization-role/index.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
+/**
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (thivi) [inline apps/console/src/features/organizations/components/edit-organization-role/index.ts:2]:
```suggestion
 * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.com) All Rights Reserved.
```

File: apps/console/src/features/organizations/components/organization-role-list.tsx [hunk 7]
```diff
@@ -0,0 +1,460 @@
+/**
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (thivi) [inline apps/console/src/features/organizations/components/organization-role-list.tsx:2]:
```suggestion
 * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.com) All Rights Reserved.
```

File: apps/console/src/features/organizations/models/index.ts [hunk 8]
```diff
@@ -18,3 +18,5 @@
 
 export * from "./endpoints";
 export * from "./organizations";
+export * from "./permission";
+
```
Comment (thivi) [inline apps/console/src/features/organizations/models/index.ts:22]: Do we need this empty line?

File: apps/console/src/features/organizations/models/permission.ts [hunk 9]
```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (thivi) [inline apps/console/src/features/organizations/models/permission.ts:2]:
```suggestion
 * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.com) All Rights Reserved.
```

**Review by thivi — APPROVED** [review]

**Review by thivi — APPROVED** [review]

File: apps/console/src/features/organizations/components/edit-organization-role/edit-organization-groups.tsx (hunk 1 above)
Comment (kavinda1995) [reply apps/console/src/features/organizations/components/edit-organization-role/edit-organization-groups.tsx]: Resolved in https://github.com/wso2/identity-apps/pull/3150/commits/4881fb75f0c759b0224acf6e192dc4addb2e9cf4

File: apps/console/src/features/organizations/components/edit-organization-role/edit-organization-permission.tsx (hunk 2 above)
Comment (kavinda1995) [reply apps/console/src/features/organizations/components/edit-organization-role/edit-organization-permission.tsx]: Resolved in https://github.com/wso2/identity-apps/pull/3150/commits/4881fb75f0c759b0224acf6e192dc4addb2e9cf4

File: apps/console/src/features/organizations/components/edit-organization-role/edit-organization-role-basic.tsx (hunk 3 above)
Comment (kavinda1995) [reply apps/console/src/features/organizations/components/edit-organization-role/edit-organization-role-basic.tsx]: Resolved in https://github.com/wso2/identity-apps/pull/3150/commits/4881fb75f0c759b0224acf6e192dc4addb2e9cf4

File: apps/console/src/features/organizations/components/edit-organization-role/edit-organization-role.tsx (hunk 5 above)
Comment (kavinda1995) [reply apps/console/src/features/organizations/components/edit-organization-role/edit-organization-role.tsx]: Resolved in https://github.com/wso2/identity-apps/pull/3150/commits/4881fb75f0c759b0224acf6e192dc4addb2e9cf4

File: apps/console/src/features/organizations/components/organization-role-list.tsx (hunk 7 above)
Comment (kavinda1995) [reply apps/console/src/features/organizations/components/organization-role-list.tsx]: Resolved in https://github.com/wso2/identity-apps/pull/3150/commits/4881fb75f0c759b0224acf6e192dc4addb2e9cf4

File: apps/console/src/features/organizations/models/permission.ts (hunk 9 above)
Comment (kavinda1995) [reply apps/console/src/features/organizations/models/permission.ts]: Resolved in https://github.com/wso2/identity-apps/pull/3150/commits/4881fb75f0c759b0224acf6e192dc4addb2e9cf4

File: apps/console/src/features/organizations/components/edit-organization-role/index.ts (hunk 6 above)
Comment (kavinda1995) [reply apps/console/src/features/organizations/components/edit-organization-role/index.ts]: Resolved in https://github.com/wso2/identity-apps/pull/3150/commits/4881fb75f0c759b0224acf6e192dc4addb2e9cf4

File: apps/console/src/features/organizations/models/index.ts (hunk 8 above)
Comment (kavinda1995) [reply apps/console/src/features/organizations/models/index.ts]: Removed in https://github.com/wso2/identity-apps/pull/3150/commits/4881fb75f0c759b0224acf6e192dc4addb2e9cf4

File: apps/console/src/features/organizations/components/edit-organization-role/edit-organization-role-users.tsx (hunk 4 above)
Comment (kavinda1995) [reply apps/console/src/features/organizations/components/edit-organization-role/edit-organization-role-users.tsx]: Resolved in https://github.com/wso2/identity-apps/pull/3150/commits/4881fb75f0c759b0224acf6e192dc4addb2e9cf4

File: apps/console/src/features/organizations/components/organization-list.tsx [hunk 10]
```diff
@@ -259,19 +278,19 @@ export const OrganizationList: FunctionComponent<OrganizationListPropsInterface>
...
-                                    { organization.ref?.length > 80 ? (
-                                        <Popup
-                                            content={ organization.ref }
-                                            trigger={ <span>{ organization.ref }</span> }
-                                        />
-                                    ) : (
-                                        organization.ref
-                                    ) }
-                                </Header.Subheader>
+                                { /*<Header.Subheader*/ }
```
Comment (yathindrakodithuwakku) [inline apps/console/src/features/organizations/components/organization-list.tsx:281]: Shall we remove the commented out code?

File: apps/console/src/features/organizations/constants/organization-constants.ts [hunk 11]
```diff
@@ -32,15 +32,20 @@ export class OrganizationManagementConstants {
...
-     * Root organization object.
+     * Super organization id as per alpha pack 2
+     */
+    public static readonly ROOT_ORGANIZATION_ID: string = "10084a8d-113f-4211-a0d5-efe36b082211";
+
+    /**
+     * Super organization object.
      *
-     * @constant
+     * @@constant
```
Comment (yathindrakodithuwakku) [inline apps/console/src/features/organizations/constants/organization-constants.ts:42]: Please remove the addition `@`

File: apps/console/src/features/organizations/components/edit-organization-role/edit-organization-groups.tsx [hunk 12]
```diff
@@ -0,0 +1,714 @@
...
+            checkedUnassignedListItems.map((role) => {
+                if (!(tempGroupList?.includes(role))) {
+                    addedRoles.push(role);
+                }
+            });
+        }
+        setTempGroupList(addedRoles);
+        setInitialTempGroupList(addedRoles);
+        setGroupList(groupList.filter(x => !addedRoles?.includes(x)));
+        setInitialGroupList(groupList.filter(x => !addedRoles?.includes(x)));
```
Comment (Rashmini) [inline apps/console/src/features/organizations/components/edit-organization-role/edit-organization-groups.tsx:255]: Let's use a meaningful name for x.

File: apps/console/src/features/organizations/components/edit-organization-role/edit-organization-groups.tsx [hunk 13]
```diff
@@ -0,0 +1,714 @@
...
+            </Modal.Actions>
+        </Modal>
+    );
+
+    const handleAssignedGroupListSearch = (e, { value }) => {
+        let isMatch = false;
+        const filteredGroupList = [];
+
+        if (!isEmpty(value)) {
+            const re = new RegExp(escapeRegExp(value), "i");
```
Comment (Rashmini) [inline apps/console/src/features/organizations/components/edit-organization-role/edit-organization-groups.tsx:552]: Shall we have a meaningful variable name?

File: apps/console/src/features/organizations/components/edit-organization-role/edit-organization-permission.tsx [hunk 14]
```diff
@@ -0,0 +1,160 @@
...
+     */
+    onRoleUpdate: () => void;
+    /**
+     * Show if the user is read only.
+     */
+    isReadOnly?: boolean;
+}
+
+/**
+ * Component to update permissions of the selected role.
```
Comment (Rashmini) [inline apps/console/src/features/organizations/components/edit-organization-role/edit-organization-permission.tsx:52]: Add new line

File: apps/console/src/features/organizations/components/edit-organization-role/edit-organization-role-basic.tsx [hunk 15]
```diff
@@ -0,0 +1,434 @@
...
+                history.push(
+                    AppConstants.getPaths().get("ORGANIZATION_ROLES")
+                );
+            }
+        });
+    };
+
+    /**
+     * Method to update role name for the selected role.
+     *
```
Comment (Rashmini) [inline apps/console/src/features/organizations/components/edit-organization-role/edit-organization-role-basic.tsx:209]: Remove unnecessary line

File: apps/console/src/features/organizations/components/edit-organization-role/edit-organization-role.tsx [hunk 16]
```diff
@@ -0,0 +1,159 @@
...
+import { useSelector } from "react-redux";
+import { RoleGroupsList } from "./edit-organization-groups";
+import { RolePermissionDetails } from "./edit-organization-permission";
+import { BasicRoleDetails } from "./edit-organization-role-basic";
+import { RoleUserDetails } from "./edit-organization-role-users";
+import { AppState, FeatureConfigInterface, history } from "../../../core";
+import { OrganizationRoleInterface } from "../../models";
+
+/**
+ * Captures props needed for edit role component
```
Comment (Rashmini) [inline apps/console/src/features/organizations/components/edit-organization-role/edit-organization-role.tsx:34]: Let's add a full-stop for the comments. Check other places as well.

File: apps/console/src/features/organizations/constants/organization-constants.ts [hunk 17]
```diff
@@ -49,3 +54,25 @@ export enum ORGANIZATION_TYPE {
     STRUCTURAL = "STRUCTURAL",
     TENANT = "TENANT"
 }
+
+
+// Role constants
```
Comment (Rashmini) [inline apps/console/src/features/organizations/constants/organization-constants.ts:59]: Check formatting

File: apps/console/src/features/organizations/pages/organization-roles-edit.tsx [hunk 18]
```diff
@@ -0,0 +1,85 @@
+import { PageLayout } from "@wso2is/react-components";
```
Comment (Rashmini) [inline apps/console/src/features/organizations/pages/organization-roles-edit.tsx:1]: Add license header

File: apps/console/src/features/organizations/pages/organization-roles-edit.tsx [hunk 19]
```diff
@@ -0,0 +1,85 @@
...
+import React, { FunctionComponent, ReactElement, useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useSelector } from "react-redux";
+import { AppConstants, AppState, FeatureConfigInterface, history } from "../../core";
+import { getOrganizationRoleById } from "../api";
+import { EditOrganizationRole } from "../components/edit-organization-role";
+import { OrganizationInterface, OrganizationRoleInterface } from "../models";
+
+const OrganizationRolesEdit: FunctionComponent = (
```
Comment (Rashmini) [inline apps/console/src/features/organizations/pages/organization-roles-edit.tsx:10]: Let's add a comment describing the component

File: apps/console/src/features/organizations/pages/organization-roles.tsx [hunk 20]
```diff
@@ -0,0 +1,492 @@
...
+    const handleListSortingStrategyOnChange = (event: SyntheticEvent<HTMLElement>, data: DropdownProps): void => {
+        setListSortingStrategy(
+            find(ORGANIZATION_ROLES_LIST_SORTING_OPTIONS, (option) => {
+                return data.value === option.value;
+            })
+        );
+    };
+
+    /**
+     *
```
Comment (Rashmini) [inline apps/console/src/features/organizations/pages/organization-roles.tsx:184]: Remove unnecessary line

File: apps/console/src/features/organizations/components/organization-list.tsx (hunk 10 above)
Comment (kavinda1995) [reply apps/console/src/features/organizations/components/organization-list.tsx:281]: Removed on later commits

File: apps/console/src/features/organizations/constants/organization-constants.ts (hunk 11 above)
Comment (kavinda1995) [reply apps/console/src/features/organizations/constants/organization-constants.ts:42]: Done in later commits

File: apps/console/src/features/organizations/components/edit-organization-role/edit-organization-groups.tsx (hunk 12 above)
Comment (kavinda1995) [reply apps/console/src/features/organizations/components/edit-organization-role/edit-organization-groups.tsx:255]: Fixed in `bd4c9bc5`

File: apps/console/src/features/organizations/components/edit-organization-role/edit-organization-groups.tsx (hunk 13 above)
Comment (kavinda1995) [reply apps/console/src/features/organizations/components/edit-organization-role/edit-organization-groups.tsx:552]: Fixed in `bd4c9bc5`

File: apps/console/src/features/organizations/components/edit-organization-role/edit-organization-permission.tsx (hunk 14 above)
Comment (kavinda1995) [reply apps/console/src/features/organizations/components/edit-organization-role/edit-organization-permission.tsx:52]: Fixed in https://github.com/wso2/identity-apps/pull/3617/commits/81488c55dd6c9c7a63575dee78e71f2b80643963

File: apps/console/src/features/organizations/components/edit-organization-role/edit-organization-role-basic.tsx (hunk 15 above)
Comment (kavinda1995) [reply apps/console/src/features/organizations/components/edit-organization-role/edit-organization-role-basic.tsx:209]: Fixed in https://github.com/wso2/identity-apps/pull/3617/commits/81488c55dd6c9c7a63575dee78e71f2b80643963

File: apps/console/src/features/organizations/components/edit-organization-role/edit-organization-role.tsx (hunk 16 above)
Comment (kavinda1995) [reply apps/console/src/features/organizations/components/edit-organization-role/edit-organization-role.tsx:34]: Fixed in https://github.com/wso2/identity-apps/pull/3617/commits/81488c55dd6c9c7a63575dee78e71f2b80643963

File: apps/console/src/features/organizations/constants/organization-constants.ts (hunk 17 above)
Comment (kavinda1995) [reply apps/console/src/features/organizations/constants/organization-constants.ts:59]: Fixed in https://github.com/wso2/identity-apps/pull/3617/commits/81488c55dd6c9c7a63575dee78e71f2b80643963

File: apps/console/src/features/organizations/pages/organization-roles-edit.tsx (hunk 19 above)
Comment (kavinda1995) [reply apps/console/src/features/organizations/pages/organization-roles-edit.tsx:10]: Fixed in https://github.com/wso2/identity-apps/pull/3617/commits/81488c55dd6c9c7a63575dee78e71f2b80643963

File: apps/console/src/features/organizations/pages/organization-roles-edit.tsx (hunk 18 above)
Comment (kavinda1995) [reply apps/console/src/features/organizations/pages/organization-roles-edit.tsx:1]: Fixed in https://github.com/wso2/identity-apps/pull/3617/commits/81488c55dd6c9c7a63575dee78e71f2b80643963

File: apps/console/src/features/organizations/pages/organization-roles.tsx (hunk 20 above)
Comment (kavinda1995) [reply apps/console/src/features/organizations/pages/organization-roles.tsx:184]: Fixed in https://github.com/wso2/identity-apps/pull/3617/commits/81488c55dd6c9c7a63575dee78e71f2b80643963

**Pull request merged**